src/llvm-multiversioning.cpp

```diff
@@ -697,6 +696,48 @@ Constant *CloneCtx::rewrite_gv_init(const Stack& stack)
     return res;
 }
 
+// replace an alias to a function with a trampoline and (uninitialized) global variable slot
+void CloneCtx::rewrite_alias(GlobalAlias *alias)
+{
+    Function *F = cast<Function>(alias->getAliasee());
+    assert(!is_vector(F->getFunctionType()));
+
+    auto Name = alias->getName();
+    alias->setName("");
+    Function *trampoline =
+        Function::Create(F->getFunctionType(), alias->getLinkage(), Name, &M);
+    trampoline->copyAttributesFrom(F);
+    alias->eraseFromParent();
+
+    uint32_t id;
+    GlobalVariable *slot;
+    std::tie(id, slot) = get_reloc_slot(F);
+
+    auto BB = BasicBlock::Create(ctx, "top", trampoline);
+    IRBuilder<> irbuilder(BB);
+
+    auto ptr = irbuilder.CreateLoad(F->getType(), slot);
+    ptr->setMetadata(llvm::LLVMContext::MD_tbaa, tbaa_const);
+    ptr->setMetadata(llvm::LLVMContext::MD_invariant_load, MDNode::get(ctx, None));
+
+    std::vector<Value *> Args;
+    for (auto &arg : trampoline->args())
+        Args.push_back(&arg);
+    auto call = irbuilder.CreateCall(ptr, makeArrayRef(Args));
+    call->setTailCallKind(CallInst::TCK_MustTail);
+    // musttail support is very bad on ARM, PPC, PPC64 (as of LLVM 3.9)
+    // Known failures includes vararg and sret.
```

Comment:
See the comment you copied. `musttail` cannot be used unconditionally (not for `sret`). It should also not be needed for non-vararg so just setting it for `vararg` functions should be fine. In other cases `tail` should be used instead.